### ISSUE TITLE
Refator Witness trait so it only require serde Serialize/Deserialize and manages hints differently

### ISSUE THREAD

**Issue by citizen-stig** [Sovereign-Labs/sovereign#263]
Discuss details with @preston-evans98 and @bkolad

**Comment by citizen-stig** [comment]
I mainly discussed it with @preston-evans98 , please take a look.

Also would be nice to have input from @bkolad @dubbelosix

**Comment by preston-evans98** [comment]
Yep, looks good for now! At some point, #246 will get fixed and remove the need to pass a `Version`s and `jmt::Node`s - but this is great for now
